### ISSUE TITLE
Does Flash Attention Support Custom Masks

### ISSUE THREAD

**Issue by Dominic23331** [Dao-AILab/flash-attention#1179]
I want to use a custom attention mask. Does flash attention support it? What should I do?

**Comment by thepowerfuldeez** [comment]
I think the best bet would be an integration with FlexAttention. Here's the example for document masking: https://github.com/pytorch-labs/attention-gym/blob/75867424a1d4391bff49527029d3612a09dd67e2/examples/flex_attn.ipynb

**Comment by JohnHerry** [comment]
What about the padding mask?

**Comment by byi8220** [comment]
Hi, has anything changed in the time since?
